fenicsxconcrete/finite_element_problem/linear_elasticity.py

```diff
@@ -68,11 +68,16 @@ def sigma(self, u):
     def solve(self, t=1.0):        
         self.displacement = self.weak_form_problem.solve()
 
+        self.compute_residual()
+
         # get sensor data
         for sensor_name in self.sensors:
             # go through all sensors and measure
             self.sensors[sensor_name].measure(self, t)
 
+    def compute_residual(self):
```

Comment:
this is then something we need to define for all (future) materials.
there should be an abstract method in the base material